h2/src/main/org/h2/mvstore/db/MVPlainTempResult.java

```diff
@@ -98,12 +104,16 @@ public boolean contains(Value[] values) {
     }
 
     private void createIndex() {
-        Builder<ValueArray, Boolean> builder = new MVMap.Builder<ValueArray, Boolean>().keyType(valueType);
+        Builder<ValueArray, Boolean> builder = new MVMap.Builder<ValueArray, Boolean>().keyType(distinctType);
         index = store.openMap("idx", builder);
-        Cursor<ValueLong, ValueArray> c = map.cursor(null);
+        Cursor<Long, ValueArray> c = map.cursor(null);
         while (c.hasNext()) {
             c.next();
-            index.putIfAbsent(c.getValue(), true);
+            ValueArray row = c.getValue();
+            if (columnCount != visibleColumnCount) {
+                row = ValueArray.get(Arrays.copyOf(row.getList(), visibleColumnCount));
+            }
+            index.putIfAbsent(row, true);
```

Comment:
suggestion: as a further optimisation, maybe this should @andreitokar 's new batch mode stuff in MVStore

Reply:
Javadoc of `append()` says
> key should be higher in map's order than any existing key

This is not guaranteed anywhere in temporary results.